compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageDescriptor.java

```diff
@@ -31,18 +34,26 @@
     private final PackageName packageName;
     private final PackageOrg packageOrg;
     private final PackageVersion packageVersion;
+    private final String repository;
```

Comment:
This should be a PackageRespository class IMO, and it should have something like DEFAULT or BUILT_IN to indicate that a dependency is compiling from a built-in repository.

Reply:
This change is not simple. Hence, I have created the issue #28988  to track the improvement. Will address it with another PR.